core/src/main/java/org/bitcoinj/quorums/FinalCommitment.java

```diff
@@ -364,7 +364,7 @@ public boolean verifySizes(LLMQParameters llmqParameters) {
     public Sha256Hash getHash() {
         try {
             UnsafeByteArrayOutputStream bos = new UnsafeByteArrayOutputStream(getMessageSize());
-            bitcoinSerializeToStream(bos, BLSScheme.isLegacyDefault());
+            bitcoinSerializeToStream(bos, isLegacy());
```

Comment:
We should serialize this object for hashing using the BLS Scheme based on the version of the object, rather than the default BLS Scheme which is a global setting.